app/components/post_body/system_message_helpers.js

```diff
@@ -0,0 +1,180 @@
+                defaultMessage: '{username} updated the channel purpose from: {oldPurpose} to: {newPurpose}',
+            };
+
+            return (
+                <Markdown
+                    baseTextStyle={messageStyle}
+                    onPostPress={onPress}
+                    textStyles={textStyles}
+                    value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+                />
+            );
+        }
+
+        localeHolder = {
+            id: 'app.channel.post_update_channel_purpose_message.updated_to',
+            defaultMessage: '{username} updated the channel purpose to: {newPurpose}',
+        };
+
+        return (
+            <Markdown
+                baseTextStyle={messageStyle}
+                onPostPress={onPress}
+                textStyles={textStyles}
+                value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+            />
+        );
+    } else if (postProps.old_purpose) {
+        localeHolder = {
+            id: 'app.channel.post_update_channel_purpose_message.removed',
+            defaultMessage: '{username} removed the channel purpose (was: {oldPurpose})',
+        };
+
+        return (
+            <Markdown
+                baseTextStyle={messageStyle}
+                onPostPress={onPress}
+                textStyles={textStyles}
+                value={formattedMessage(localeHolder, {username, oldPurpose, newPurpose}, intl)}
+            />
+        );
+    }
+
+    return null;
+};
+
+const renderDisplayNameChangeMessage = (postBodyProps, styles, intl) => {
+    const {onPress, postProps} = postBodyProps;
+    const {messageStyle, textStyles} = styles;
+    const oldDisplayName = postProps.old_displayname;
+    const newDisplayName = postProps.new_displayname;
+
+    if (!(postProps.username && postProps.old_displayname && postProps.new_displayname)) {
+        return null;
+    }
+
+    const username = renderUsername(postProps.username);
+    const localeHolder = {
+        id: 'api.channel.post_update_channel_displayname_message_and_forget.updated_from',
+        defaultMessage: '{username} updated the channel display name from: {oldDisplayName} to: {newDisplayName}',
+    };
+
+    return (
+        <Markdown
+            baseTextStyle={messageStyle}
+            onPostPress={onPress}
+            textStyles={textStyles}
+            value={formattedMessage(localeHolder, {username, oldDisplayName, newDisplayName}, intl)}
+        />
+    );
+};
+
+const systemMessageRenderers = {
+    [Posts.POST_TYPES.HEADER_CHANGE]: renderHeaderChangeMessage,
+    [Posts.POST_TYPES.DISPLAYNAME_CHANGE]: renderDisplayNameChangeMessage,
+    [Posts.POST_TYPES.PURPOSE_CHANGE]: renderPurposeChangeMessage,
+};
+
+export const renderSystemMessage = (postBodyProps, styles, intl) => {
+    const renderer = systemMessageRenderers[postBodyProps.postType];
+    if (!renderer) {
```

Comment:
by only rendering this when the post is an actual system message you don't need this if statement

Reply:
Hmm.. unfortunately this check is still needed because `isSystemMessage` isn't indicative of all the various 'system' messages, as @hmhealey mentioned in another review comment.

>I don't think isSystemMessage can be used here because we won't have a renderer for every system message type like ephemeral messages or anything new that gets added later on.